### PR TITLE
Shell input

### PR DESCRIPTION
**Pull request by joshinils**
see if you can compile

### CODE REVIEW

**Comment by philipp-zettl** [comment]
```
~/wirdnix/util/Shell.cpp: In constructor ‘Message::Message(const string&, olc::Pixel, uint32_t)’:
~/wirdnix/util/Shell.cpp:32:5: error: ‘localtime_s’ was not declared in this scope
     localtime_s(&buf, &t);
     ^~~~~~~~~~~
~/wirdnix/util/Shell.cpp:32:5: note: suggested alternative: ‘localtime_r’
     localtime_s(&buf, &t);
     ^~~~~~~~~~~
     localtime_r
```

File: util/Shell.cpp [hunk 1]
```diff
@@ -0,0 +1,299 @@
+#include "Shell.h"
+#include "Engine.h"
+#include <time.h>
```
Comment (philsupertramp) [inline util/Shell.cpp]: Found [solution](https://en.cppreference.com/w/c/chrono/localtime). 
>As with all bounds-checked functions, localtime_s is only guaranteed to be available if __STDC_LIB_EXT1__ is defined by the implementation and if the user defines __STDC_WANT_LIB_EXT1__ to the integer constant 1 before including time.h.

Comment (philsupertramp) [inline util/Shell.cpp]:
```suggestion
#define __STDC_WANT_LIB_EXT1__ 1
#include "Shell.h"
#include "Engine.h"
#include <time.h>
```

File: util/Shell.cpp [hunk 2]
```diff
@@ -0,0 +1,299 @@
+#include "Shell.h"
```
Comment (philsupertramp) [inline util/Shell.cpp]:
```suggestion

```

File: util/Shell.cpp [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+#include "Engine.h"
```
Comment (philsupertramp) [inline util/Shell.cpp]:
```suggestion

```

File: util/Shell.cpp [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+    , _messageWidth(Engine::ScreenWidth() / (12 * _scale))
+    , _message(generateMessageString(_originalMessage))
+    // recalculate _displayWidth, as initialization of static member is too early
+    , _messageSprite(_displayWidth = Engine::ScreenWidth() * 2 / 3 + 2 + Message::PADDING, _lines * 8 * _scale + 2)
+    , _color(color)
+    , _age(0)
+    , _backgroundColor(BACKGROUND_COLOR)
+    // 12 = 1.5(2/3 screen with) * 8(default font with)
+{
+    initSprite();
```
Comment (philsupertramp) [inline util/Shell.cpp]:
```suggestion
#ifdef __STDC_LIB_EXT1__
    initSprite();
    time_t t = time(NULL);
    struct tm buf;
    localtime_s(&buf, &t);
    buf.tm_sec += 20;
    _createdAt = mktime(&buf);
#endif
```

**Pull request merged**